### PR TITLE
SDKS-6291: User consent config

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Added UserConsent config using a property wrapper
- Replaced old impressions mode property by wrapper (Transparent to the user)
- Added tests for wrappers and config functioning for Swift and Objc

### CODE REVIEW

File: Split/Api/SplitClientConfig.swift [hunk 1]
```diff
@@ -231,6 +221,19 @@ public class SplitClientConfig: NSObject {
     ///
     @objc public var synchronizeInBackground = false
 
+    /// Indicates if events and impressions are tracked and sent to BE.
+    /// @param mode Values:<br>
+    ///     UNKNOWN: Impressions and events are tracked in memory. They are lost on destroy
+    ///     GRANTED: Impressions and events are tracked and sent. They remains stored when the SDK is destroyed.
```
Comment (gthea) [inline Split/Api/SplitClientConfig.swift:227]:
```suggestion
    ///     GRANTED: Impressions and events are tracked and sent. They remain stored when the SDK is destroyed.
```

File: Split/Api/SplitClientConfig.swift [hunk 2]
```diff
@@ -231,6 +221,19 @@ public class SplitClientConfig: NSObject {
...
+    ///     many times impressions are posted. If previous value was UNKNOWN, tracked data is persisted.
```
Comment (gthea) [inline Split/Api/SplitClientConfig.swift:228]: Just checking; is the `many times impressions are posted` sentence OK?

File: Split/Storage/Impressions/ImpressionsMode.swift [hunk 3]
```diff
@@ -24,3 +24,39 @@ enum ImpressionsMode: String {
...
+    public init(wrappedValue: String) {
+        setValue(wrappedValue)
+    }
+
+    private mutating func setValue(_ newValue: String) {
+        let uppercased = newValue.uppercased()
+        guard let projectedValue = ImpressionsMode(rawValue: uppercased) else {
+                Logger.w("You passed an invalid impressionsMode (\(uppercased)), " +
+                    " impressionsMode should be one of the following values: " +
+                            "'DEBUG', 'OPTIMIZED' or 'NONE'. Defaulting to 'OPTMIZED' mode.")
```
Comment (gthea) [inline Split/Storage/Impressions/ImpressionsMode.swift:51]:
```suggestion
                            "'DEBUG', 'OPTIMIZED' or 'NONE'. Defaulting to 'OPTIMIZED' mode.")
```

**Review by gthea — APPROVED** [review]

File: Split/Api/SplitClientConfig.swift (hunk 2 above)
Comment (javrudsky) [reply Split/Api/SplitClientConfig.swift:228]: mmm not really :S Thanks @gthea! will change that in next PR

File: Split/Storage/Impressions/ImpressionsMode.swift (hunk 3 above)
Comment (javrudsky) [reply Split/Storage/Impressions/ImpressionsMode.swift:51]: Thanks @gthea! will change that in next PR

**Pull request merged**